Create Issue counter Integration Test

Areiser opened the pull request:
This test implements an integration test for the issue counter in the core jenkins column.

Areiser commented:
I apologize for the messy PR, everything should be fine now, I can rebase if necessary.
I didn't know where to place the text files containing the warnings for this integration test, so if there is a better folder please let me know.

uhafner commented on src/test/java/io/jenkins/plugins/analysis/core/columns/IssueCounterITest.java:
```diff
@@ -0,0 +1,94 @@
+ */
+public class IssueCounterITest extends IntegrationTestWithJenkinsPerTest {
+
+    private static final String ISSUE_ELEMENT_XPATH = "/html/body/div[4]/div[2]/div[2]/div[2]/table/tbody/tr[2]/td[8]/a";
```
Is this path stable? Wouldn't it be easier if we add a div with ID? I added one for your, see https://github.com/jenkinsci/warnings-ng-plugin/commit/bfab4b43725b97801eab4c9b2dc6724af048829b

uhafner commented on src/test/java/io/jenkins/plugins/analysis/core/columns/IssueCounterITest.java:
```diff
@@ -0,0 +1,94 @@
+
+    /**
+     * Tests that the issue counter is displayed correctly with 1 issue.
+     * @throws IOException Shouldn't throw.
```
Remove `@throws` and extract the method that throws the exception. There you can catch it and rethrow a RTE.

uhafner commented on src/test/java/io/jenkins/plugins/analysis/core/columns/IssueCounterITest.java:
```diff
@@ -0,0 +1,94 @@
+    public void shouldShowIssueCounterEqualToOne() throws IOException {
+        createAndBuildProjectWithFile(JAVAC_ONE_WARNING);
+
+        List<DomNode> element = getRootPage().getByXPath(ISSUE_ELEMENT_XPATH);
```
Extract into method. It would also make sense that the method returns a list of Strings?

uhafner commented on src/test/java/io/jenkins/plugins/analysis/core/columns/IssueCounterITest.java:
```diff
@@ -0,0 +1,94 @@
+     * @throws IOException Shouldn't throw.
+     */
+    @Test
+    public void shouldShowIssueCounterEqualToTen() throws IOException {
```
This test is the same as the test above, remove it.

uhafner commented on src/test/java/io/jenkins/plugins/analysis/core/columns/IssueCounterITest.java:
```diff
@@ -0,0 +1,94 @@
+        enableWarnings(project, new Java());
+        scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+
+        List<DomNode> element = getRootPage().getByXPath(ISSUE_ELEMENT_XPATH);
```
Is there a way to check for the health report?

Areiser replied:
I'm not sure what the goal is if this test is for the issue counter column. Should I check for the health status as well? 
Do you mean the hover menu for different tools? If so, I can access it by XPath as well but it might not be stable.

uhafner replied:
Yes, I mean the hover, sorry, not the health report. 
It should be part of the DOM, or isn't it?

uhafner commented on src/test/java/io/jenkins/plugins/analysis/core/columns/IssueCounterITest.java:
```diff
@@ -0,0 +1,94 @@
+
+        assertThat(Integer.parseInt(element.get(0).asText())).isEqualTo(0);
+    }
+
```
I think the following tests are missing: 
- `-` as result.
- Multiple tools so that the column actually renders the sum. 
- Tool selection via `getJenkins().getInstance().getPrimaryView().getColumns()...`

Areiser replied:
I added a test for the dash case and multiple tools.
I'm not sure what you mean with the tool selection, what exactly should I test with it? If the selection is possible via UI or test it again using the "non-ui" way?

uhafner replied:
I think you can have three parts in your test `shouldShowIssueCounterEqualToSum`. 
1. The existing test
2. Change the visible tools to `java` using the API call (non UI) `getJenkins().getInstance().getPrimaryView().getColumns()...` and validate the number of warnings
3. Change the visible tools to `python` using the API call (non UI) `getJenkins().getInstance().getPrimaryView().getColumns()...` and validate the number of warnings

Maybe it would make sense to change the number of warnings for python or java to 2 so that the assertion correctly works

Areiser replied:
As discussed, this functionality is not working for the test.

Areiser commented:
I added a test for the hover part to check for the correct values for different tools in the hover healthreport and rebased the branch.
I think this should cover the hover functionality and the sum test.
